File: Player.py
import sys


class Point:

    def __init__(self, x=-1, y=-1):
        self.x, self.y = x, y


class Player:
    id = -1
    is_me = False

    def __init__(self, nbr_drones):
        self.drones = [Drone() for i in xrange(nbr_drones)]


class Drone:
    nbr_instances = 0
    p = Point()
    
    def __init__(self):
        Drone.nbr_instances += 1
        
    def __del__(self):
        Drone.nbr_instances -= 1
        
    
class Zone:
    
    def __init__(self, centre=None):
        self.centre = centre if centre else Point()


# Read init information from standard input, if any
players = drones = Zones = None
def init():
    global players, drones, zones
    nbr_players, id, nbr_drones, nbr_zones = [int(x) for x in raw_input().split()]
    
    #create players
    players = [Player(nbr_drones) for i in xrange(nbr_players)]
    players[0].is_me, players[0].id = True, id
    
    #create zones
    zones = [Zone(Point(*[int(x) for x in raw_input().split()])) for i in xrange(nbr_zones)]

    
init()

while 1:
    #do your stuff
    pass

File: Player.dart
import 'dart:io';

var players, drones, zones, my_id, nbr_drones;

class Zone {
    int x;
    int y;
    int etat;
    
    Zone(this.x, this.y, [this.etat = -1]);
    
    String toString() => "$x $y $etat";
}

class Drone {
    int x;
    int y;
    
    Drone(this.x, this.y);
    
    Drone.origin(){
        x = 0;
        y = 0;
    }
    
    String toString() => "$x $y";
}
    

class Player {
    List<Drone> drones = <Drone>[];
    
    Player();
    
    String toString() => "$drones";
}

void init() {
    
    var infos = readValues();
    var nbr_players = infos[0];
    my_id = infos[1];
    nbr_drones = infos[2];
    var nbr_zones = infos[3];
    
    zones = new List.generate(nbr_zones,(_){
        var positions = readValues();
        return new Zone(positions[0], positions[1]);
    });
    
    players = new List.generate(nbr_players, (_) =>
      new Player()..drones = new List.generate(nbr_drones, (_) => new Position.origin())
    );

}

void printZones() =>
    zones.forEach(stderr.writeln); 


void printPlayer() =>
    players.forEach(stderr.writeln);

List<int> readValues() =>
    stdin.readLineSync().split(' ').map(int.parse).toList();

void main() {
    init();
    
    while (true) {
        zones.forEach((Zone z) => z.etat = readValues()[0]);
        
        players.forEach((player) =>
            player.drones.forEach((drone) {
                var positions = readValues();
                drone..x = positions[0]
                     ..y = positions[1]; 
            })
        );

        //  What to do?
        
        //  Print new pos for my drone
        players[my_id].drones.forEach((_) => print("3999 1799"));
    }
  
}